megamek/src/megamek/common/MoveStep.java

```diff
@@ -2403,47 +2403,41 @@ && getClearance() > 0) {
                 } else {
                     movementType = EntityMovementType.MOVE_RUN;
                 }
-            } else if (canUseSprint(game)
-                    && ((getMpUsed() <= sprintMPNoMASC)
-                            || ((getMpUsed() <= sprintMP) && (isMASCUsedEntity || isSuperchargerUsedEntity)))
-                    && !isRunProhibited() && !isEvading()) {
-                if (entity.getMovementMode() == EntityMovementMode.VTOL
-                        || (entity.getMovementMode() == EntityMovementMode.WIGE
-                                && getClearance() > 0)) {
-                    movementType = EntityMovementType.MOVE_VTOL_SPRINT;
-                } else {
-                    movementType = EntityMovementType.MOVE_SPRINT;
-                }
-            } else if ((getMpUsed() <= sprintMPOneMASC) && !isRunProhibited()
-                    && !isEvading()) {
-                // decide if using MASC or Supercharger if need only one
-                // choose Super if even
-                MPBoosters mpBoosters = entity.getArmedMPBoosters();
-                int scTarget = mpBoosters.hasSupercharger() ? entity.getSuperchargerTarget() : 2000;
-                int mascTarget = mpBoosters.hasMASC() ? entity.getMASCTarget() : 2000;
-                if (mascTarget < scTarget) {
+            } else if (canUseSprint(game)) {
+                if (((getMpUsed() <= sprintMPNoMASC) || ((getMpUsed() <= sprintMP) && (isMASCUsedEntity || isSuperchargerUsedEntity))) && !isRunProhibited() && !isEvading()) {
```

Comment:
I feel like this if statement series is complicated enough that it could use a comment explaining the "real life" scenarios we're dealing with.